interceptor.go

```diff
@@ -48,7 +48,8 @@ type UnaryServerInfo struct {
 }
 
 // UnaryHandler defines the handler invoked by UnaryServerInterceptor to complete the normal
-// execution of a unary RPC.
+// execution of a unary RPC. All errors returned by StreamHandler are compatible with the
```

Comment:
This is something users implement (their service handler).  So:

"are compatible with" -> "should be produced by"

Also, please add something like: "Otherwise, the status code returned will be codes.Unknown."

(same for StreamHandler)

Reply:
👍  on 'produced by'.

However, regarding the second part - I'm confused by the "otherwise". Aren't all the errors (including the `codes.Unknown` ones I touched) "produced" by the status package via `status.New`? As in, in the first sentence you talk about errors, but the second you talk about codes.

Reply:
This may clarify (add to godoc string too?):

The error returned by this handler determines the status code and message of the RPC.  If the error was produced by the status package, the code and message encoded in the error will be used directly.  Otherwise, the status code will be UNKNOWN and the status message will be the string returned by error's Error method.

Reply:
Ahh ok - I think I see what you're going for. Would something like this make sense?

> All errors returned by UnaryHandler should either be produced by status package, or wrapped in a status.Status with codes.Unknown as the status code and err.Error() as the status message.

^ trying to condense

Otherwise, I'm happy to write it as you have it!

Reply:
> All errors returned by UnaryHandler will either be produced by the status package,
> or wrapped in a status.Status with codes.Unknown as the status code and err.Error()
> as the status message.

Maybe I wasn't clear on what this interface is for.  A `UnaryHandler` is something the _user_ implements (it's their server's RPC handler), and _we_ call.  This doc string should tell them what they need to do.  If they return another error type, that's fine, but we'll turn their error into an UNKNOWN status code for the RPC when we send the trailers back to the client.  This does not describe types of things we will return from the function.  (Your version seems to be describing something we are in control of.)  How about this slight modification:

`All errors returned by a UnaryHandler implementation should be produced by the status package, or else gRPC will use codes.Unknown as the status code and err.Error() as the status message of the RPC.`

Maybe an additional sentence like "a nil error indicates success", even though it's obvious, since the above may lead someone to think they always have to return some error, even on successes, or else they'll get an UNKNOWN error code.  Or reword the first sentence to "If a UnaryHandler returns an error, it should be produced by ....". (?)

Reply:
>  A UnaryHandler is something the user implements (it's their server's RPC handler), and we call

AH shoot that's the part I was entirely missing. Thanks! Incoming update.